### PR TITLE
allow sensible-defaults/use-all-settings to take some exceptions

### PR DESCRIPTION
**Pull request by tzz**
This requires CL currently (`loop` macro), I can rewrite it with something else.

It allows people to accept all the defaults except some.

If this is acceptable I'll update the docs as well.

### CODE REVIEW

**Comment by tzz** [comment]
ping @hrs ?

**Comment by hrs** [comment]
Hey, thanks @tzz! And sorry for the slow reply. I like this idea!

I'm concerned that, because we're filtering functions by name instead of just calling them, we're restricting ourselves to only calling functions that don't have any required arguments (as you noticed in the case of `set-default-line-length-to`). That might not be a problem in practice, though, since we're currently using default values when we call functions from `use-all-settings`. So I think I've argued myself into being fine with it. :smile:

File: sensible-defaults.el [hunk 1]
```diff
@@ -162,31 +162,35 @@ frame just for that file."
...
+                   sensible-defaults/make-dired-file-sizes-human-readable
+                   sensible-defaults/shorten-yes-or-no
+                   sensible-defaults/always-highlight-code
+                   sensible-defaults/refresh-buffers-when-files-change
+                   sensible-defaults/show-matching-parens
+                   sensible-defaults/flash-screen-instead-of-ringing-bell
+                   sensible-defaults/set-default-line-length-to
+                   sensible-defaults/open-clicked-files-in-same-frame-on-mac
+                   sensible-defaults/yank-to-point-on-mouse-click
+                   )
```
Comment (hrs) [inline sensible-defaults.el:191]: This list of functions has become data. Maybe we should extract it into a constant:

``` emacs
(defconst *sensible-defaults/default-functions*
  '(sensible-defaults/open-files-from-home-directory
    ;; ...
    sensible-defaults/yank-to-point-on-mouse-click))
```

Then we can loop with `(loop for f in *sensible-defaults/default-functions* ...`, which reads nicely to me.

**Comment by tzz** [comment]
I don't like either approach so far because it has to keep two lists in sync manually :)

What about simply tagging the functions and then searching for any function that matches the package prefix and the tag? That's easy to implement.

Another approach is to make every sensible-default a macro that records itself in the list variable. I also like that approach because a macro could potentially allow every sensible-default to be wrapped with undo functionality, activation/deactivation logic, metadata about its effects, and even dependencies (you can't use X if you're disabling Y).

If you like any approach so far enough, I can give implementing it a try.

**Pull request closed without merging**